lib/util.js

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+function base64UrlToBase64(b64u) {
+  return b64u.replace(/\-/g, '+').replace(/_/g, '/');
+}
+
+function base64UrlToString(b64u) {
+  var b64 = base64UrlToBase64(b64u);
+  switch (b64.length % 4) {
+    case 0:
+      break;
+    case 2:
+      b64 += '==';
+      break;
+    case 3:
+      b64 += '=';
+      break;
+    default:
+      throw 'Not a valid Base64Url';
+  }
+  var utf8 = atob(b64);
+  try {
+    return decodeURIComponent(escape(utf8));
+  } catch (e) {
+    return utf8;
+  }
+}
+
+function stringToBuffer(str) {
+  var res = new Uint8Array(str.length);
+  for (var i = 0; i < str.length; i++) {
+    res[i] = str.charCodeAt(i);
+  }
+  return res;
+}
+
+function base64UrlDecode(str) {
+  return atob(base64UrlToBase64(str));
+}
+
+module.exports = {
```

Comment:
It's closer to the [ES6 standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) than AMD. We'll ultimately be moving to ES6, so the transition will be easier.